### ISSUE TITLE
[Enhancements] Ingame User Configuration  options for advanced setups

### ISSUE THREAD

**Issue by mikeloeven** [refinedmods/rangedpumps#31]
I started learning how to use this mod mostly due to the fact it has become such a prevalent in mod packs as a replacement for build craft pumps and a less leggy alternative to the Ender-Thermic pump. It excels in performance and basic functionality however that is also the main issue the fact that the pump has no user configurable parameters outside of an external configuration file. I would like to see an in game UI for the pump that provides Filtering and scan area parameters. 

- Filtering is simple toggle between Water Only Lava Only Both I did see a previously closed issue with someone expressing concern about pumps placed over an ocean randomly sucking up lava and destroying machines. It was closed as works as intended which frankly even if it IS intended its not behavior that any sane or logical player would accept as intended and thus especially for a mod trying to improve performance setting off chain reactions of machine explosions by randomly dumping lava into a water pipe is counter productive and could lead to lag spikes or server crashes. Filtering is needed in order to keep the mod relevant. You also dont want to encourage users to go back to using the older server lagging monstrosities.

- Geometry settings. A UI for setting range  scan offset and I would recommend an auto scan mode for contiguous source blocks. Config file settings would be used for upper / lower limits for the per pump settings  

- Scan offset The Offset shifts the scan area around the pump itself this allows configurations anywhere between centered on the pump with 64blocks in each direction or 128 blocks in a single direction / diagonally useful for placing pumps on a shore where source blocks expand in one direction and scanning around the pump in a square would end up not producing meaningful amounts of fluid due to wasting ticks scanning rock. 
- Range X Y Z range limits (yes get that y range in there) self explanatory in game user defined dimensions of the scan cube within limits set by the config. 
- Contiguous source blocks ( This is an auto configuration option for quick setup on a ocean or lava pool. Instead of using a brute force block by block scan the scan will try to map out the Shoreline out to the range limitation and restrict its scan area to to the desired reservoir of source blocks and will not try to scan for blocks outside the pool its sitting in. 

In conclusion it would be highly beneficial to include In game settings for fine tuning pump behaviors which i believe will take the mod to the next level.
